projects/netgrif-components-core/src/lib/navigation/breadcrumbs/abstract-breadcrumbs.component.ts

```diff
@@ -1,41 +1,128 @@
+import {SimpleFilter} from "../../filter/models/simple-filter";
+import {take} from "rxjs/operators";
+import {BehaviorSubject, Subscription} from "rxjs";
+import {ActivatedRoute, Router} from "@angular/router";
+import {
+    DynamicNavigationRouteProviderService
+} from "../../routing/dynamic-navigation-route-provider/dynamic-navigation-route-provider.service";
+import {Case} from "../../resources/interface/case";
+import {I18nFieldValue} from "../../data-fields/i18n-field/models/i18n-field-value";
+import {TranslateService} from "@ngx-translate/core";
 
 @Component({
     selector: 'ncc-breadcrumbs-component',
     template: ''
 })
-export abstract class AbstractBreadcrumbsComponent {
+export abstract class AbstractBreadcrumbsComponent implements OnDestroy, AfterViewInit {
 
     @Input() showHome: boolean = true;
+    @Input() showFilter: boolean = true;
+    @Input() redirectOnClick: boolean = true;
     @Input() lengthOfPath: number = 30;
     @Input() partsAfterDots: number = 2;
+    filterName: string;
+    breadcrumbsParts: Array<string>;
     private static DOTS: string = '...';
     private static DELIMETER: string = '/';
+    private static NODE_PATH: string = 'nodePath';
+    private static ITEM_SETTINGS: string = 'item_settings';
     private _showPaths: boolean = false;
+    private nicePath: BehaviorSubject<Array<string>>;
+    private redirectUrls: Map<string, Array<string>>;
+    private nicePathSubscription: Subscription;
 
-    protected constructor(protected _uriService: UriService) {
+    protected constructor(protected _uriService: UriService,
+                          protected _caseResourceService: CaseResourceService,
+                          protected _activatedRoute: ActivatedRoute,
+                          protected _router: Router,
+                          protected _dynamicRoutingService: DynamicNavigationRouteProviderService,
+                          protected _translateService: TranslateService) {
+        this.nicePath = new BehaviorSubject<Array<string>>(undefined);
+        this.redirectUrls = new Map<string, Array<string>>();
+        this.initNicePath();
     }
 
-    public getPath(): Array<string> {
-        const tmp = this._uriService.splitNodePath(this._uriService.activeNode);
-        if (tmp?.length > this.partsAfterDots + 1 && this._uriService.activeNode?.uriPath.length > this.lengthOfPath && !this._showPaths) {
-            const newPath = [tmp[0], AbstractBreadcrumbsComponent.DOTS];
-            for (let i = tmp.length - this.partsAfterDots; i < tmp.length; i++) {
-                newPath.push(tmp[i]);
+    ngAfterViewInit() {
+        this.resolveBreadcrumbs();
+    }
+
+    ngOnDestroy(): void {
+        if (!!this.nicePathSubscription) {
+            this.nicePathSubscription.unsubscribe();
+        }
+    }
+
+    public resolveBreadcrumbs() {
+        const filterId = this._activatedRoute.snapshot.params.filterCaseId
```

Comment:
nejaky log ?

if (!filterId || !this._uriService.activeNode) {
    logger.error('Missing required data for resolving breadcrumbs.');
    return;
}

Reply:
Done